From 267e91ae95bcf44a668ad76ff2b7284bc48cc57e Mon Sep 17 00:00:00 2001
From: Roberto Granados <x0095451@ti.com>
Date: Fri, 4 Mar 2011 16:52:10 -0600
Subject: OMAP4: DSS: Clear BURSTTYPE bit when TILER is not used

A kernel L3 error was found in a case where a hardware overlay
was used with both a Tiler buffer and a linear buffer. The error
occurs when a linear buffer is attached to a hardware overlay
after a Tiler buffer had been attached.
BURSTTYPE bit was not being reset back to 0 for subsequent
non-Tiler cases.

Change-Id: Iaa7ab0de8f97356af859307155a191f38eaf5709
Signed-off-by: Steven Fischer <steven.fischer@motorola.com>
Signed-off-by: Roberto Granados Dorado <x0095451@ti.com>
---
 drivers/video/omap2/dss/dispc.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

Index: kernel/drivers/video/omap2/dss/dispc.c
===================================================================
--- kernel.orig/drivers/video/omap2/dss/dispc.c	2011-04-08 12:33:09.000000000 +0900
+++ kernel/drivers/video/omap2/dss/dispc.c	2011-04-08 16:01:17.710863379 +0900
@@ -3212,6 +3212,9 @@
 	/* Reset the Tiler burst bit */
 	REG_FLD_MOD(dispc_reg_att[plane], 0x0, 29, 29); /*BURSTTYPE */
 
+	/* Reset the Tiler burst bit */
+	REG_FLD_MOD(dispc_reg_att[plane], 0x0, 29, 29); /*BURSTTYPE */
+
 	if (rotation_type == OMAP_DSS_ROT_TILER) {
 #ifdef CONFIG_TILER_OMAP
 		int bpp = color_mode_to_bpp(color_mode) / 8;
